home-service/src/main/java/com/softserveinc/ita/homeproject/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,75 @@
+package com.softserveinc.ita.homeproject.service.impl;
+
+import com.softserveinc.ita.homeproject.homedata.entity.User;
+import com.softserveinc.ita.homeproject.homedata.repository.RoleRepository;
+import com.softserveinc.ita.homeproject.homedata.repository.UserRepository;
+import com.softserveinc.ita.homeproject.model.CreateUser;
+import com.softserveinc.ita.homeproject.model.ReadUser;
+import com.softserveinc.ita.homeproject.model.UpdateUser;
+import com.softserveinc.ita.homeproject.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+
+    private final UserRepository userRepository;
+    private final ConversionService conversionService;
+    private final RoleRepository roleRepository;
+
+    @Transactional
+    @Override
+    public ReadUser createUser(CreateUser payload) {
+        if (userRepository.findByEmail(payload.getEmail()).isPresent()) {
+            throw new RuntimeException();  //TODO: here throw already exist exception
```

Comment:
Resolve TODO

Reply:
Should I make "exceptions" task in this branch?